assignment-2/lean.md

```diff
@@ -1,3 +1,147 @@
+period and effort and, although they produce deliverables and artefacts,
+they are not the requested one by the client. This qualifies as *muda*
+type I, as it is a waste, but it is necessary. This waste does not exist
+in scrum. In scrum, the documentation is almost not existing or it is
+very reduced based in the Agile Manifesto directives \[2\]. This means
+that the team does not produce any other deliverables but the product
+that the client requests.
+
+RUP also has more phases besides development. The early phases focus on
+having clear requirements and late ones focus on deployment and testing.
+These phases are also present in scrum but they happen in a different
+way. In scrum, part of the requirements are gathered by the product
+owner from the clients and the other part is done by the developers
+during the sprints. The testing and deployment are also integrated
+during the sprint instead in a separate phase. In RUP, the division of
+the phases aims to reduce the risks in the project by specifying the
+details of the project ahead and making sure that the product has been
+properly deployed and tested. Although necessary, these phases add extra
+waste in terms of processes and management, instead of being integrated
+within the development. In scrum, this waste does not exist because it
+is considered part of the development and it does not carry extra
+overhead due to being integrated into the development phase.
+
+In RUP, the construction phase is done by a requirements team which is
+composed by requirement engineers who are not the developers who will
+implement the software during the implementation phase. This practice is
+suboptimal because it leads to a situation in which there are
+intermediaries and artefacts between the clients and the team that will
+implement the product. In scrum, although the requirements are gathered
+by the *product owner*, the development team which will implement the
+product is involved in the creation of the backlog.
+
+On the other hand, scrum can be more wasteful than RUP. Scrum works with
+smalls teams and big projects require more than one scrum team working
+on it. This means that those teams would have each one their own *scrum
+master* who will act as facilitator and a *product owner* who will talk
+with the clients. This situation will lead to role duplicity which is
+not optimal at all. It is true that given the case, the scrum master and
+the product owner can be shared between different teams but if teams
+belong to different organisations or divisions will not happen. RUP only
+has one management team which is an improvement in terms of waste over
+scrum.
+
+Scrum also focuses on small iterations of the product: features are
+added iteration after iteration. The client then will not receive the
+final product requested during the first iterations, which is a waste
+because it is not what was requested. We understand as an *incomplete
```

Comment:
Small iterations is a waste compared to what? having the final product in one sprint? Having nothing at all?
Do note, that RUP is also an iterative approach.